--- a/pynicotine/search.py
+++ b/pynicotine/search.py
@@ -463,6 +463,14 @@
             return
 
         username = msg.username
+
+        if search.mode == "user":
+            expected_users = search.users or ()
+
+            if username not in expected_users:
+                msg.token = None
+                return
+
         ip_address, _port = msg.addr
 
         if core.network_filter.is_user_ignored(username):
