--- a/pynicotine/search.py
+++ b/pynicotine/search.py
@@ -627,6 +627,13 @@
             msg.token = None
             return
 
+        if search.mode == "user":
+            expected_users = search.users or ()
+
+            if username not in expected_users:
+                msg.token = None
+                return
+
         if isinstance(search, WishSearchRequest) and (search.is_ignored or username in search.ignored_users):
             msg.token = None
             return
